### ISSUE TITLE
Prediction 

### ISSUE THREAD

**Issue by caglarerennn** [rvorias/ind_knn_ad#4]
-How can i make a prediction for an image?
-How can i train the model with my own data?
Thanks.

**Comment by rvorias** [comment]
Hi!

The simplest thing to do is to arrange your data as a MVTec dataset.
Here is an example of how it's done: https://github.com/rvorias/ind_knn_ad#custom-datasets
You can also look at the datasets that the repo automatically downloads.

If you want to predict something, I've also added a little snippet to the readme:

```python
from indad.model import SPADE

model = SPADE(k=42, backbone="hypernet")

# feed healthy dataset
model.fit(...)

# get predictions
img_lvl_anom_score, pxl_lvl_anom_score = model.predict(...)
```

The first `...` is a dataset. Like I've said before, it's easy when you use the MVTec dataset structure
The second `...` is a sample, so a 4-d torch tensor of size [1, 3, 224, 224].

Hope that helps!

Btw, scores are not normalized, you will need to define your own threshold based on your required prec/recall.

**Comment by caglarerennn** [comment]
### I get an output image but it doesn't give a result similar to the examples :(        

**model.fit(train_ds)
PATH = "/home/Desktop/DL/AnomalyDetection/ind_knn_ad-master/model/"+str(cls)+".pth"
torch.save(model.state_dict(), PATH)

print("One image testing ...")

pil_img = Image.open("/home/Desktop/DL/AnomalyDetection/ind_knn_ad-master/datasets/carpet/test/cut/009.png")
pil_img = pil_img.resize((224, 224))
pil_to_tensor = transforms.ToTensor()(pil_img).unsqueeze_(0)
img_lvl_anom_score, pxl_lvl_anom_score = model.predict(pil_to_tensor)
trans = torchvision.transforms.ToPILImage()
out = trans(pxl_lvl_anom_score)
#print(pxl_lvl_anom_score)
out.save("asd.png")**

**Comment by rvorias** [comment]
You are missing some normalization.

Try this:
```python
out = trans(pxl_lvl_anom_score[0]/torch.max(pxl_lvl_anom_score))
```
Your image will be in greyscale though.

If you want to visualize it in a notebook, you can also work with matplotlib:
```python
import matplotlib.pyplot as plt

plt.imshow(pxl_lvl_anom_score[0])
```

You can also work with an overlay effect, e.g.:
```python
def save_image(sample, fmap):
    timestamp = datetime.now().strftime("%d_%m_%Y_%H_%M_%S")

    plt.imshow(sample.permute([1,2,0]))
    plt.imshow(fmap, cmap='jet', alpha=0.5)
    plt.gca().set_axis_off()
    plt.subplots_adjust(top = 1, bottom = 0, right = 1, left = 0, 
                hspace = 0, wspace = 0)
    plt.margins(0,0)
    plt.gca().xaxis.set_major_locator(plt.NullLocator())
    plt.gca().yaxis.set_major_locator(plt.NullLocator())
    plt.savefig(f"results/image_{timestamp}.png", bbox_inches = 'tight',
        pad_inches = 0)
```

Gonna close this issue soon as it has nothing to do with the repo.
